### PR TITLE
parser: Fix infinite loop when parsing generic type

### LINKED ISSUE #1287: Cast array.size() in if comparison spins at 100% CPU
To reproduce:

```
function main() {
  let arr: [u8] = []
  if arr.size() as! i64 < 0 { }
}
```

### PR DESCRIPTION
**Pull request by SimonRask**
Fixes #1287

### CODE REVIEW

**Review by alimpfard — COMMENTED** [review]
Could you also add a test to make sure this doesn't break in the future?

**Comment by SimonRask** [comment]
I have now added a test for the reduced version of the code provided in #1287.

Im not sure if the code provided should be valid, but the test ensures that the parser doesn't stall for that case.

File: selfhost/parser.jakt [hunk 1]
```diff
@@ -3398,7 +3398,13 @@ struct Parser {
                 if .current() is LessThan {
                     .index++
                     while not .current() is GreaterThan and not .eof() {
-                        params.push(.parse_typename())
+                        let parsed_type_parameter = .parse_typename() 
+                        if parsed_type_parameter is Empty {
+                            .error("Expected type inside generic", .current().span())
```
Comment (alimpfard) [inline selfhost/parser.jakt:3403]: :thinking: Could we be a bit more helpful with this error? something like
> Error: Expected a type here (pointing after the `<`)
> Hint: This type forms a generic instantiation (pointing at the `i64 <` range)

**Comment by alimpfard** [comment]
Also, please rebase on top of main, CI will be unhappy otherwise.

**Pull request closed without merging**